mne/io/edf/edf.py

```diff
@@ -572,7 +571,28 @@ def _read_edf_header(fname, exclude):
                 edf_info['units'].append(1e-6)
             else:
                 edf_info['units'].append(1)
+
         ch_names = [ch_names[idx] for idx in sel]
+        units = [units[idx] for idx in sel]
+        # make sure channel names are unique
+        # code modified from _check_consistency function in meas_info.py
+        unique_ids = np.unique(ch_names, return_index=True)[1]
+        if len(unique_ids) != len(ch_names):
+            dups = set(ch_names[x]
+                       for x in np.setdiff1d(range(len(ch_names)), unique_ids))
+            warn('Channel names are not unique, found duplicates for: '
+                 '%s. Applying running numbers for duplicates.' % dups)
+            for ch_stem in dups:
+                overlaps = np.where(np.array(ch_names) == ch_stem)[0]
+                n_keep = min(len(ch_stem),
+                             14 - int(np.ceil(np.log10(len(overlaps)))))
+                ch_stem = ch_stem[:n_keep]
+                for idx, ch_idx in enumerate(overlaps):
+                    ch_name = ch_stem + '-%s' % idx
+                    assert ch_name not in ch_names
```

Comment:
I'd replace the assert statement with a try/except block.

Reply:
I disagree. In this case, `assert` is much easier to read. But I'm not sure if this falls in cases that we allow `assert` inside the code. We had a discussion once about that. I should find it and if these questions keep popping up we should uniform the code write something about it in the contribution guide.

Reply:
but if `assert` is not allowed I would use `if` case and raise.

Reply:
Yes, `raise` is fine. You cannot rely on `assert` to handle runtime errors, it is a debugging aid. In this case I would argue that the condition could in principle occur, so we should raise an error. With `assert`, you document that a condition will never fail (otherwise there's a bug in the code) - which is not what we want here IMO.